### PR TITLE
Space Polyp Milking Fix and Gelatin Condiment Sprites

### PR DESCRIPTION
**Pull request by Optimism333**
## Jelly Bugfix Stuff ##
- Space polyps can now be milked without also spilling the reagent container on them, addressing: #33189 
- Code for the coin trading polyp subtype has been cleaned up

## Tiny Bites of Jelly Content ##
polyp_condiment
- Polyps now make some visible emotes periodically
- Polyp gelatin now has a unique condiment bottle sprite and a sprite overlay

[bugfix] [content]

:cl:
 * bugfix: Polyps can be milked without spilling the liquid on them every time you click on them with the bucket
 * bugfix: Cleaned up the code for the coin trading polyp subtype, and removed the duplicated milking interaction
 * rscadd: Polyps have some visible emotes
 * rscadd: Polyp gelatin now has a condiment bottle sprite and condiment overlay

### CODE REVIEW

**Comment by Optimism333** [comment]
Sprites for a polyp gelatin condiment bottle and condiment overlay have been added.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -39,6 +39,7 @@
 		/obj/machinery/apiary,
 		/mob/living/simple_animal/cow,
 		/mob/living/simple_animal/hostile/retaliate/goat,
+		/mob/living/simple_animal/hostile/retaliate/polyp,
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/glass.dm]: do that thing we discussed

and at mob level make it intent based

Comment (Optimism333) [reply code/modules/reagents/reagent_containers/glass.dm]: I'll give it a shot. No guarantees, though. I've never tried writing a proc before.

File: code/modules/mob/living/simple_animal/hostile/mothership_creatures.dm [hunk 2]
```diff
@@ -264,6 +272,19 @@
 				to_chat(user, "<span class='warning'>[O] is full.</span>")
 			if(!transfered)
 				to_chat(user, "<span class='warning'>[src]'s tendrils are dry. Wait a bit longer...</span>")
+		if(trades_coins == 1 && istype(O, /obj/item/weapon/reagent_containers/food/snacks/zam_notraisins))
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/mothership_creatures.dm:278]: also while you're at it, this is interpreted the same as `if(trades_coins && ...)`
You don't need to compare it to 1, you just need to make sure it's non-zero - this is purely style though

File: code/modules/reagents/reagent_containers/glass.dm (hunk 1 above)
Comment (Optimism333) [reply code/modules/reagents/reagent_containers/glass.dm]: Well, an attempt was made.

**Comment by Optimism333** [comment]
Tried to apply Kurf's proc idea. Still very WIP

**Comment by Optimism333** [comment]
These two pages have a can_be_placed_into = null

code\game\objects\items\robot\robot_items\robot_replenishing_container.dm
code\modules\detectivework\footprints_and_rag.dm

Therefore the deletion of the var list has caused problems. Would appreciate advice on what to do here.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 3]
```diff
@@ -77,7 +44,7 @@
 	if (!adjacency_flag)
 		return
 
-	if (is_type_in_list(target, can_be_placed_into))
+	if (target == !splashable)
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/glass.dm]: splashable()

Comment (Optimism333) [reply code/modules/reagents/reagent_containers/glass.dm]: Fix here applied. Thanks

**Comment by Optimism333** [comment]
At least it complies now.

**Comment by Kurfursten** [comment]
Looks good

**Comment by Optimism333** [comment]
> Looks good

Something is still broken somewhere. Mobs are getting splashed still.

broke

Going to be pretty busy today, but I'll see if I can figure out what went wrong in the evening.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 4]
```diff
@@ -77,7 +44,7 @@
 	if (!adjacency_flag)
 		return
 
-	if (is_type_in_list(target, can_be_placed_into))
+	if (target == !splashable())
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/glass.dm]: Oh, I can't believe I overlooked this before, very amusing

`if(!target.splashable())`

You need to be calling the splashable() proc on the target, not comparing to see if the target is the result of this object's splashable proc

Comment (Optimism333) [reply code/modules/reagents/reagent_containers/glass.dm]: I'll fix this when I get home in the evening

Comment (Optimism333) [reply code/modules/reagents/reagent_containers/glass.dm]: Fixed!

**Comment by Optimism333** [comment]
The proc is functional thanks to Kurf. I did a bit of testing and it didn't seem to have any further issues.

Marking this as ready for review.

**Comment by Optimism333** [comment]
Well, I did find one bit of funny behavior, actually. You can now insert a rag into a chemistry dispenser to directly add reagents.

**Comment by Kurfursten** [comment]
You can explicitly stop this by adding it in chem_dispenser.dm L582

Just add these lines:
```
if(istype(I, /obj/item/weapon/reagent_containers/glass/rag))
  return FALSE
```

Not the cleanest, but it solves the problem. On the other hand, directly dispensing into a rag isn't... really a big deal either.

**Comment by Optimism333** [comment]
> You can explicitly stop this by adding it in chem_dispenser.dm L582
> 
> Just add these lines:
> 
> ```
> if(istype(I, /obj/item/weapon/reagent_containers/glass/rag))
>   return FALSE
> ```
> 
> Not the cleanest, but it solves the problem. On the other hand, directly dispensing into a rag isn't... really a big deal either.

Will keep your fix in my notes in case any weird behavior with other machinery turns up, but yeah, I don't think that causes a massive break in either mechanics or spess logic.

I'm fine with leaving it as is for now.

**Pull request merged**